apps/adresse-service/src/main/java/no/nav/adresse/service/consumer/command/PdlAdresseSoekCommand.java

```diff
@@ -23,27 +25,28 @@
     private final String token;
 
     @Override
-    public PdlAdresseResponse call() {
-
-        try {
-            return webClient
-                    .post()
-                    .uri(builder -> builder.path("/pdl-api/graphql").build())
-                    .body(BodyInserters.fromValue(query))
-                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
-                    .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
-                    .header(TEMA, TemaGrunnlag.GEN.name())
-                    .retrieve()
-                    .bodyToMono(PdlAdresseResponse.class)
-                    .block();
-
-        } catch (
-                WebClientResponseException e) {
-            log.error("Feil ved henting av adressedata: {}.", e.getResponseBodyAsString());
-            log.error(query.toString());
-            throw e;
-        }
+    public Mono<PdlAdresseResponse> call() {
+
+        return webClient
+                .post()
+                .uri(builder -> builder.path("/pdl-api/graphql").build())
+                .body(BodyInserters.fromValue(query))
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+                .header(TEMA, TemaGrunnlag.GEN.name())
+                .exchange()
+                .flatMap(response -> response.bodyToMono(PdlAdresseResponse.class)
+                        .map(value -> {
+                            if (!value.getErrors().isEmpty()) {
+                                throw new BadRequestException("Spørring inneholder feil: " + value.getErrors().toString());
```

Comment:
For å ikke bryte promise kallene må du wrappe exceptions inn i `Mono.error(....)`

`Mono.error(throw new BadRequestException("Spørring inneholder feil: " + value.getErrors().toString()));`